demo/end2end_image_demo.py

```diff
@@ -0,0 +1,102 @@
+from argparse import ArgumentParser
+
+import mmcv
+
+from mmdet.apis import init_detector
+from mmocr.apis.inference import model_inference
+from mmocr.core.end2end_visualize import end2end_show_result, write_json
+from mmocr.core.prepare_model import prepare_det_model, prepare_recog_model
+from mmocr.datasets.pipelines.crop import crop_img
+from mmocr.models import build_detector  # noqa: F401
+
+
+def end2end_inference(args, det_model, recog_model):
+    image_path = args.img
+    end2end_res = {'filename': image_path}
+    end2end_res['result'] = []
+
+    image = mmcv.imread(image_path)
+    det_result = model_inference(det_model, image)
+    bboxes = det_result['boundary_result']
+
+    for bbox in bboxes:
+        box_res = {}
+        box_res['box'] = [round(x) for x in bbox[:8]]
+        box_res['box_score'] = float('{:.4f}'.format(bbox[8]))
+        box_img = crop_img(image, bbox[:8])
+
+        if args.recog_model == 'crnn':
+            box_img = mmcv.bgr2gray(box_img, keepdim=True)
+
+        recog_result = model_inference(recog_model, box_img)
+
+        text = recog_result['text']
+        text_score = recog_result['score']
+        if isinstance(text_score, list):
+            text_score = sum(text_score) / max(1, len(text))
+        box_res['text'] = text
+        box_res['text_score'] = float('{:.4f}'.format(text_score))
+        end2end_res['result'].append(box_res)
+
+    return end2end_res
+
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument('img', type=str, help='Input Image file.')
+    parser.add_argument(
+        'out_file', type=str, help='Output file name of the visualized image.')
+    parser.add_argument(
+        '--detect-model',
+        type=str,
+        default='psenet',
+        choices=['psenet', 'panet', 'dbnet', 'textsnake', 'maskrcnn'],
+        help='Text detect model type.')
+    parser.add_argument(
```

Comment:
I suggest to add 5 more argument so that user can specify any config and model they want to test.

--detect-config
--detect-ckpt
--recog-config
--recog-ckpt
--recog-rgb (whether use rgb in reognition model)

if the above arguments is set, then we use the model user specified. Otherwise, we use --recog-alg
and --detect-alg

Reply:
Yeah pass configs instead of algorithm names